block/manager.go

```diff
@@ -174,12 +199,22 @@ func getAddress(key crypto.PrivKey) ([]byte, error) {
 	return cmcrypto.AddressHash(rawKey), nil
 }
 
+// SetBlockStore is used to set Block Store used by Manager.
+func (m *Manager) SetBlockStore(blockStore *goheaderstore.Store[*types.Block]) {
```

Comment:
I think you can delete this method, because you only use it once when creating a fullnode and you can avoid needing it by simply creating the exchange service first, and then passing that into the `NewManager` function.

Otherwise we need to create a top level manager mutex to protect setting these fields since this method could be called randomly and it would be a potential race condition if the blockstore was being accessed.

Reply:
Made the change you suggested here: https://github.com/rollkit/rollkit/pull/1128 but leads to breaking the CI tests right now. Is there anything I'm missing here?

Reply:
left a comment, you aren't assigning the field 👍